java/server/src/org/openqa/grid/web/servlet/api/v1/Proxy.java

```diff
@@ -39,7 +41,11 @@
     }
 
     final String proxyToFind = query.replaceAll("^/", "");
-    final RemoteProxy proxy = this.getRegistry().getAllProxies().getProxyById(proxyToFind);
+    RemoteProxy proxy = this.getRegistry().getAllProxies().getProxyById(proxyToFind);
```

Comment:
> In that case the earlier change itself was sufficient, because it was basically querying based on whatever the user passes. So we can revert it back.

Well, that makes the HTTP GET URL look silly again. I'll think about / bounce the consideration off of others a little more .. 

>> Just thinking aloud here. I am not sure if this is something that should be left to the user to customize because this is internally used by the node to uniquely identify itself (which you called out). In that case, should a user really fiddle around with this 

The hub registration protocol makes these allotments.  There are use cases where 'autodetecting' or 'assuming' the proxy id is not sufficient. In Se2.x the allotments were also there -- except you could only specify proxy `id` via code or the config json, if I recall correctly.